controllers/cloud.redhat.com/namespaces.go

```diff
@@ -142,8 +155,10 @@ func GetReadyNamespaces(ctx context.Context, cl client.Client) ([]core.Namespace
 	for _, ns := range nsList.Items {
 		for _, owner := range ns.GetOwnerReferences() {
 			if owner.Kind == "NamespacePool" {
-				if val, ok := ns.ObjectMeta.Annotations["env-status"]; ok && val == "ready" {
-					ready = append(ready, ns)
+				if val := ns.ObjectMeta.Labels["pool"]; val == pool {
```

Comment:
In the follow up PR lets break this out into a new function

Reply:
Will create a new branch to implement this change after this branch is merged.